assets/src/blocks/TopicLink/TopicLinkEditor.js

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable no-shadow */
+
+const {useSelect} = wp.data;
+const {
+  BlockControls,
+  MediaUpload,
+  MediaUploadCheck,
+  InspectorControls,
+} = wp.blockEditor;
+const {
+  SelectControl,
+  PanelBody,
+  ToolbarGroup,
+  ToolbarButton,
+  FocalPointPicker,
+} = wp.components;
+const {sprintf, __} = wp.i18n;
+
+/**
+ * TopicLinkEditor component for managing the Topic Link block in the editor.
+ *
+ * @param {Object} props - Component properties.
+ * @param {Object} props.attributes - Block attributes.
+ * @param {boolean} props.isSelected - Indicates if the block is selected.
+ * @param {Function} props.setAttributes - Function to update block attributes.
+ * @return {JSX.Element} The Topic Link Editor component.
+ */
+export const TopicLinkEditor = ({
+  attributes,
+  isSelected,
+  setAttributes,
+}) => {
+  const {
+    categoryId,
+    categoryName,
+    focal_points,
+    imageId: customImageId,
+    imageUrl: customImageFromId,
+  } = attributes;
+
+  const {
+    categoriesList,
+    imageId,
+    imageUrl,
+    imageAlt,
+    currentPostCategories,
+  } = useSelect(select => {
+    const categoriesList = [].concat(
+      select('core').getEntityRecords('taxonomy', 'category', {
+        hide_empty: true,
+        per_page: -1,
+      }) || []
+    );
+
+    const customImage = customImageId && select('core').getMedia(customImageId);
+    const customImageFromId = customImage?.source_url;
+
+    const imageId = customImageId;
+    const imageUrl = customImageFromId;
+    const imageAlt = customImage?.alt_text;
+    const currentPostCategories = select('core/editor').getCurrentPost().categories || [];
+
+    return {
+      categoriesList,
+      imageId,
+      imageUrl,
+      imageAlt,
+      currentPostCategories,
+    };
+  }, [categoryId, customImageId, customImageFromId]);
+
+  if (!categoriesList.length) {
+    return __('Populating block\'s fields…', 'planet4-blocks-backend');
+  }
+
+  /**
+   * Sets the block's category based on available categories or current post categories.
+   */
+  const setBlockCategory = () => {
+    const postCategory = categoriesList.find(category => category.id === currentPostCategories[0]);
+    const blockCategory = categoriesList.find(category => category.id === categoryId);
+
+    let categoryData = null;
```

Comment:
What do you think to set as `let categoryData = categoriesList[0];` if it after set within the [`else` statement](https://github.com/greenpeace/planet4-master-theme/pull/2470/files#diff-52a02a9810dbc2c3f222b8c0c3bbbfd3abe68f9f43717bf14f616494d8e26054R92)?

Reply:
I don't know if I understand your comment correctly.
If you are suggesting something like this:
``` 
const setBlockCategory = () => {
    const postCategory = categoriesList.find(category => category.id === currentPostCategories[0]);
    const blockCategory = categoriesList.find(category => category.id === categoryId);

    let categoryData = categoriesList[0];
    setAttributes({categoryId: parseInt(categoriesList[0].id)});

    if (blockCategory) {
      categoryData = blockCategory;
      setAttributes({categoryId: parseInt(blockCategory.id)});
    } else if (postCategory) {
      categoryData = postCategory;
      setAttributes({categoryId: parseInt(postCategory.id)});
    }

    setAttributes({categoryLink: categoryData?.link || ''});
    setAttributes({categoryName: categoryData?.name || ''});
  };
``` 
Let me let you know that it crashes the editor:

``` 
Warning: Maximum update depth exceeded. This can happen when a component calls setState inside useEffect, but useEffect either doesn't have a dependency array, or one of the dependencies changes on every render.
```

Reply:
Nope! What I'm suggesting is 

```suggestion
   let categoryData = categoriesList[0];

    if (blockCategory) {
      categoryData = blockCategory;
    } else if (postCategory) {
      categoryData = postCategory;
   } 
   
    setAttributes({categoryId: parseInt(categoryData.id)});
```

What do you think?

Reply:
Oh, I see!
I like it, I just applied it.